common/setups/hybrid/gmm_system.py

```diff
@@ -0,0 +1,878 @@
+        gmm_sat_args: GmmSatArgs,
+        gmm_vtln_sat_args: GmmVtlnSatArgs,
+        train_data: Dict[str, GmmDataInput],
+        dev_data: Dict[str, GmmDataInput],
+        test_data: Dict[str, GmmDataInput],
+    ):
+        """
+        :param gmm_init_args: parameters for am and feature extraction
+        :param gmm_monophone_args: parameters for lin align and monophone
+        :param gmm_triphone_args: parameters for cart and triphone
+        :param gmm_vtln_args: parameters for VTLN
+        :param gmm_sat_args: parameters for SAT
+        :param gmm_vtln_sat_args: parameters for VTLN+SAT
+        :param dict[str, GmmInput] train_data:
+        :param dict[str, GmmInput] dev_data:
+        :param dict[str, GmmInput] test_data:
+        :return:
+        """
+        self.gmm_init_args = gmm_init_args
+        self.gmm_monophone_args = gmm_monophone_args
+        self.gmm_triphone_args = gmm_triphone_args
+        self.gmm_vtln_args = gmm_vtln_args
+        self.gmm_sat_args = gmm_sat_args
+        self.gmm_vtln_sat_args = gmm_vtln_sat_args
+
+        self._init_am(**self.gmm_init_args.am_args)
+        for name, v in sorted(train_data.items()):
+            self.add_corpus(name, corpus=v, add_lm=False)
+            self.train_corpora.append(name)
+
+        for name, v in sorted(dev_data.items()):
+            self.add_corpus(name, corpus=v, add_lm=True)
+            self.dev_corpora.append(name)
+
+        for name, v in sorted(test_data.items()):
+            self.add_corpus(name, corpus=v, add_lm=True)
+            self.test_corpora.append(name)
+
+        self.cart_questions = self.gmm_triphone_args_args.cart_questions
+
+    @tk.block()
+    def _init_am(self, **kwargs):
+        self.crp["base"].acoustic_model_config = am.acoustic_model_config(**kwargs)
+        allow_zero_weights = kwargs.get("allow_zero_weights", False)
+        if allow_zero_weights:
+            self.crp["base"].acoustic_model_config.mixture_sew.allow_zero_weights = True
+            self.crp[
+                "base"
+            ].acoustic_model_config.old_mixture_set.allow_zero_weights = True
+
+    @tk.block()
+    def _init_corpus(self, name):
+        segm_corpus_job = corpus_recipes.SegmentCorpusJob(
+            self.corpora[name].corpus_file, self.concurrent[name]
+        )
+        self.set_corpus(
+            name=name,
+            corpus=self.corpora[name],
+            concurrent=self.concurrent[name],
+            segment_path=segm_corpus_job.segment_path,
+        )
+        self.jobs[name]["segment_corpus"] = segm_corpus_job
+
+    @tk.block()
+    def _init_lm(self, name, file, type, scale, **kwargs):
+        self.crp[name].language_model_config = rasr.RasrConfig()
+        self.crp[name].language_model_config.type = type
+        self.crp[name].language_model_config.file = file
+        self.crp[name].language_model_config.scale = scale
+
+    @tk.block()
+    def _init_lexicon(self, name, file, normalize_pronunciation, **kwargs):
```

Comment:
I would prefer to more consistently document all types in parameters.

I assume `name` is `str` (this is most often the case).

I assume `normalize_pronunciation` is `bool`, although I'm not 100% sure.

But I have no idea what `file` is. Is this `io.TextIO` (accepting strings), or `io.RawIO` (accepting bytes), or a filename, or maybe a `Path`, or sth else? (Btw, for filenames, I usually would prefer `filename` as an argument name, to make that more clear.)